block/src/Pos/Block/Logic/Header.hs

```diff
@@ -384,13 +399,13 @@ getHeadersRange depthLimitM older newer = runExceptT $ do
     -- Proving newerD >= olderD
     let newerOlderF = "newer: "%build%", older: "%build
     when (newerD == olderD) $
-        throwError $
-        sformat ("getHeadersRange: newer and older headers have "%
+        throwGHR $
+        sformat ("newer and older headers have "%
                  "the same difficulty, but are not equal. "%newerOlderF)
                 newerHd olderHd
     when (newerD < olderD) $
-        throwError $
-        sformat ("getHeadersRange: newer header is less dificult than older one. "%
+        throwGHR $
```

Comment:
This and the `throwGHR` on line 402 seem misplaced. Why are doing these checks here? Surely this function is taking on too much responsibility. Why can't it just fetch the headers and leave it at that?

Reply:
Well, underlying traversal functions require me to put in some sane data, so this is a perfect place to check check its validity. `getHashesRange` is _actually_ a listener, and it's not used anywhere else, so the only difference between doing checks here and doing them outside is how do we handle errors. Here there's `GetHashesRangeException` already that will carry all the info about bad user request. If we do checks outside, we need yet another exception type. Also these checks are simple/fast.

Reply:
> getHashesRange is actually a listener, and it's not used anywhere else, so the only difference between doing checks here and doing them outside is how do we handle errors

I disagree. Sure its only user at the moment is a listener, but it has no intrinsic relation to networking / listener / diffusion layer. One could argue it's a database component but even that is a bit dubious. `getHashesRange` is what its type says it is: given two hashes, get (somehow) a list of hashes. Doesn't matter where the inputs came from, doesn't matter where the output goes.